arrabbiata/src/columns.rs

```diff
@@ -38,16 +40,21 @@ pub enum Column {
 
 /// Convert a column to a usize. This is used by the library [mvpoly] when we
 /// need to compute the cross-terms.
-/// For now, we only support the conversion of the private inputs, i.e. the `X`
-/// variant.
-/// Also, for now, the [mvpoly] library expects the columns to be mapped to a
-/// prime number as the prime representation is used under the hood. This must
-/// be changed.
+/// For now, only the private inputs and the public inputs are converted,
+/// because there might not need to treat the selectors in the polynomial while
+/// computing the cross-terms (FIXME: check this later, but pretty sure it's the
+/// case).
+///
+/// Also, the [mvpoly::monomials] implementation of the trait [mvpoly::MVPoly]
+/// will be used, and the mapping here is consistent with the one expected by
+/// this implementation, i.e. we simply map to an increasing number starting at
+/// 0, without any gap.
 impl From<Column> for usize {
     fn from(val: Column) -> usize {
         match val {
             Column::X(i) => i,
-            _ => unimplemented!("Only the private inputs are supported for now"),
+            Column::PublicInput(i) => NUMBER_OF_COLUMNS + i,
```

Comment:
Doesn't having (at least) a full columns for PI can create issue on the perfs of the verifier.
More importantely,  this translate in the folder verifier which will handle a big PI.
We can use the PLONK way to handle PI to avoid a PI being linear in the number of rows

Since this is not the topic of the PR, we can see that later

Reply:
Yes, I want to investigate other solutions to handle PI.